src/pages/Login.js

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useRef } from "react";
+import { UserContext, CredContext } from "../common/contexts";
+import Form from "../components/Form";
+import { useHistory } from "react-router-dom";
+import { Card, CardLink } from "../styles/Card";
+import { Span } from "../styles/Span";
+
+function Login() {
+  const { setUser } = useContext(UserContext);
+  const { creds } = useContext(CredContext);
+  const history = useHistory();
+  const errorRef = useRef("");
+
+  const login = ({ email, password }) => {
+    let cred = creds.find(
+      (cred) => cred.email === email && cred.password === password
+    );
+    if (cred !== undefined) {
+      setUser({ email, password, auth: true });
+      history.push({
+        pathname: "/home",
+      });
+    } else {
+      errorRef.current.textContent = "Email or password do not match";
```

Comment:
Probably the use of ref is not optimum here. You could have the errors as reusable styled components.